File: yh_star/src/yh_star_pub.cpp
'''손생님풀이'''
#include "ros/ros.h"
#include "yh_star/yh_star_msg.h"

int main(int argc, char** argv)
{
    ros::init(argc, argv, "yh_star_pub");
    ros::NodeHandle nh;

    ros::Publisher yh_star_pub = nh.advertise<yh_star::yh_star_msg>("yh_star_topic",100);

    ros::Rate loop_rate(2); //0.5초

    yh_star::yh_star_msg msg; //보낼 메세지

    int cnt = 0;

    while(ros::ok())   //오케이 인동안
    {
        msg.stamp = ros::Time::now()
        msg.data = cnt;
        pub.publish(msg)
        cnt++;
        loop_rate.sleep();
    
    }

    return 0;

}

File: yh_star/yh_star_scripts/yh_star_pub.py
#!/usr/bin/python
# -*- coding: utf-8 -*-


""" 필수 상황

#!/usr/bin/python
# -*- coding: utf-8 -*-
한글을 쓰려면 이렇게 주석 처리 해야함. 
 """
import rospy
from yh_star.msg import yh_star_msg

def talker():
   
    pub = rospy.Publisher("yh_star_topic",yh_star_msg,queue_size=100)
    rospy.init_node("yh_star_pub",anonymous=True)
    loop_rate = rospy.Rate(2)
    cnt = 0
    msg = yh_star_msg()


    while not rospy.is_shutdown():
        msg.stamp = rospy.get_rostime()
        msg.data = cnt 
        '''rospy.loginfo("send message: %d", msg.stamp.secs)
        rospy.loginfo("send message: %d", msg.stamp.nsecs)
        rospy.loginfo("send message: %d", msg.data)'''

        cnt += 1
        pub.publish(msg)
        loop_rate.sleep() 
        
if __name__=="__main__":
    try:
        talker()
    except rospy.ROSInterruptException:
        pass